File: senz-client-samples/ruby/sender.rb
require_relative 'utils/AesUtils'
require_relative 'utils/imageUtils'
require_relative 'client'

sharedKey  = "LGWlhb329Y09oluI"
imagePathToSend = "sample.jpg"

#to register device
puts "registering a device"
timeS = timeStamp()
regmsg = "SHARE #pubkey KEY @senz #time %s ^dev1 signature" % [timeS]
sendMessage(regmsg)

#convert image to byte string
byteString = imageToString(imagePathToSend)

#encrypting byte string using AES
aes = AesUtil.new(sharedKey)
byteString =  aes.encrypt(byteString)

#sending Message
timeS = timeStamp()
msg = "DATA $image %s @dev2 #time %s ^dev1 signature" % [byteString,timeS]
puts "do you want to send message(y/n)"
ans = gets.chomp
if(ans == 'y')
  puts "sending message"
  sendMessage(msg)
end

File: senz-client-samples/java/Sender.java
import java.util.Scanner;

public class Sender {
  static String deviceName = "dev1";
  static String sharedKey =  "LGWlhb329Y09oluI";
  static String imagePathToSend = "sample.jpg";

  public static void main(String args[]) {
    Client client = new Client();
    Image_Utils imageUtils = new Image_Utils();
    AES_Utils aesUtils;
    Scanner sc = new Scanner(System.in);

    // Register device
    System.out.println("Registering device...");
    String msg = "SHARE #pubkey key @senz #time "+client.getTimeStamp()+" ^"+deviceName+" signature";
    client.sendMessage(msg);

    // Convert image to byte string
    String byteString = imageUtils.imageToString(imagePathToSend);

    // Encrypt using AES Crypto
    aesUtils = new AES_Utils(sharedKey);
    byteString = aesUtils.encrypt(byteString);

    // Send the message
    System.out.println("Press enter to send the image...");
    sc.nextLine();

    msg = "DATA $image " + byteString +" @dev2 #time " + client.getTimeStamp() + " ^dev1 signature";
    client.sendMessage(msg);

    // Unregister device
    System.out.println("Unregistering device...");
    msg = "UNSHARE #pubkey key @senz #time " + client.getTimeStamp() +" ^"+ deviceName +" signature";
    client.sendMessage(msg);
    sc.close();
  }
}
